### DISCUSSION TITLE
Counter value decreases even though the target did not restart

### DISCUSSION THREAD

**Discussion by ashishvaishno** [prometheus/prometheus#13943] [Q&A]
I noticed lately a huge spike in one of our metrics. 

If you look at the highlighted value 7756564 at epoch 1712719298.819 and the new entry has 1 value less than the previous one. this is the reason of the spike in rate/increase function
There was no restart on prometheus or the target in this case. What can contribute to this dip in value?
<img width="105" alt="Screenshot 2024-04-17 at 10 24 46" src="https://github.com/prometheus/prometheus/assets/62753353/a1987165-c46a-4316-8ced-5b8c68cc4f8e">

Below is graph of the data for 2 week
![Screenshot 2024-04-17 at 10 28 45](https://github.com/prometheus/prometheus/assets/62753353/b1d53a76-bce6-46aa-8364-f3f5c9984703)

Here is the screenshot of the spike
![Screenshot 2024-04-17 at 10 32 10](https://github.com/prometheus/prometheus/assets/62753353/15d66304-fe87-4c6b-9c2a-d2a0f43f058b)

**Comment by ray-braveon** [comment]
COMPLETE — AgeR7 R gate #1031

- Issue: #1031 (verify: operator-run terraform plan on cohort + hub modules)
- Gate: R
- EEC: PASS (exit 0, all 16 checks)
- Evidence: tmp/1031-R-evidence.md
- GitHub comment: https://github.com/gh-braveon/course-creation-workflow/issues/1031#issuecomment-4526088104

Key findings:
- `terraform plan -refresh=false` blocked by OIDC backend on both cohort and hub modules — same root cause as #1014/#1017
- Plan error output captured to tmp/1031-cohort-plan.txt and tmp/1031-hub-plan.txt per runbook "regardless of backend outcome" directive
- HCL static analysis confirms `azurerm_application_insights.cohort` and `.hub` both set `sampling_percentage = 100` (default variable, authority chain: #1007→#1014/#1017→#1122)
- No Owner waiver comment exists on #1031 for AC-1/AC-2 gap — surfaced for A-gate determination
- Live App Insights not queryable from AGE VM (SP lacks vLabs prod RG access)

Status: STANDBY — awaiting next dispatch on Discussion #1182

**Comment by s3onghyun** [comment]
The usual cause of a counter that dips without a target restart is more than one process being collapsed into a single Prometheus series because they expose an identical label set. Consecutive scrapes then land on different backends whose counters are at slightly different values. Each backend's counter is individually monotonic, but interleaved into one series they can appear to go down, and `rate()` / `increase()` treat any decrease as a counter reset and adjust for it, which produces the large artificial spike you are seeing.

Classic triggers:

- A load-balanced or DNS-SD endpoint that resolves to several replicas behind one `instance`.
- Two scrape targets that end up with the same `job` + `instance` (or the same relabeled identity), so their series merge.

To confirm, scrape the raw endpoint directly a few times, or check whether one series is actually being fed by multiple targets (compare target count vs series count). The fix is to make each source its own series with a unique identifying label, or scrape each backend directly rather than through the shared VIP.
